core-library/src/main/java/org/silverpeas/core/util/media/streaming/StreamingProvider.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2000 - 2021 Silverpeas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * As a special exception to the terms and conditions of version 3.0 of
+ * the GPL, you may redistribute this Program in connection with Free/Libre
+ * Open Source Software ("FLOSS") applications as described in Silverpeas's
+ * FLOSS exception.  You should have received a copy of the text describing
+ * the FLOSS exception, and it is also available here:
+ * "https://www.silverpeas.org/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.silverpeas.core.util.media.streaming;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Data defining a stream provider.
+ * @author silveryocha
+ */
+public class StreamingProvider implements Serializable {
+  private static final long serialVersionUID = 3725979193939989226L;
+
+  private final String name;
+  private final Pattern urlIdExtractorRegexpPattern;
+  private final String oembedUrlStringPattern;
+  private final boolean justIdInOembedUrl;
+  private final List<Pattern> regexpDetectionParts;
+
+  StreamingProvider(final String name, final Pattern urlIdExtractorRegexpPattern,
+      final String oembedUrlStringPattern, final boolean justIdInOembedUrl,
+      final List<Pattern> regexpDetectionParts) {
+    this.name = name;
+    this.urlIdExtractorRegexpPattern = urlIdExtractorRegexpPattern;
+    this.oembedUrlStringPattern = oembedUrlStringPattern;
+    this.justIdInOembedUrl = justIdInOembedUrl;
+    this.regexpDetectionParts = regexpDetectionParts;
+  }
+
+  /**
+   * This constructor is provided for test environments.
+   * <p>
+   *   It is not usable. It permits to avoid technical errors.
+   * </p>
+   * @param name a name.
+   */
+  public StreamingProvider(final String name) {
+    this(name, null, null, false, null);
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  String getOembedUrlStringPattern() {
+    return oembedUrlStringPattern;
+  }
+
+  public List<Pattern> getRegexpDetectionParts() {
+    return regexpDetectionParts;
+  }
+
+  /**
+   * Gets the identifier of the stream from an url.
+   * @param url a stream URL
+   * @return an optional stream identifier.
+   */
+  Optional<String> extractStreamingId(String url) {
```

Comment:
Parameters defined into property file will be explained as best as can do. So here, there is no need to explain how the id is extracted from full URL of a streaming .
The scope is package private because of its technical use.